packages/eslint-plugin/src/rules/no-redundant-type-constituents.ts

```diff
@@ -323,8 +327,58 @@ export default createRule({
               }
             }
           }
-        }
 
+          typePartFlags.length >= 2
+            ? seenUnionTypes.set(typeNode, typePartFlags)
+            : null;
+        }
+        const checkIfUnionsAreAssignable = (): undefined => {
+          let typeFlagsOfUnions: ts.TypeFlags[] = [];
+          let result = true;
+          let primitiveUnit: number;
+
+          seenUnionTypes.forEach((value, key) => {
```

Comment:
[Style] This code is already kind of complex, let's stick with informative names.

```suggestion
          seenUnionTypes.forEach((unionTypeFlags, unionTypeNode) => {
```

(or some names with more meaning - I'm not sold on those)